### PR TITLE
BZ2187603: SCC and SA for DPDK checkup

### PR DESCRIPTION
**Pull request by apinnick**
Version(s): 4.13

Issue: [BZ2187603](https://bugzilla.redhat.com/show_bug.cgi?id=2187603)

Link to docs preview: 
- [Monitoring overview](https://58871--docspreview.netlify.app/openshift-enterprise/latest/virt/support/monitoring/virt-monitoring-overview.html)
- [VM latency checkup](https://58871--docspreview.netlify.app/openshift-enterprise/latest/virt/support/monitoring/virt-running-cluster-checkups.html#virt-measuring-latency-vm-secondary-network_virt-running-cluster-checkups)
- [DPDK checkup](https://58871--docspreview.netlify.app/openshift-enterprise/latest/virt/support/monitoring/virt-running-cluster-checkups.html#virt-checking-cluster-dpdk-readiness_virt-running-cluster-checkups)

QE review:
- [x] QE has approved this change.

Additional information: PR includes general doc housekeeping changes.

### CODE REVIEW

File: modules/virt-checking-cluster-dpdk-readiness.adoc [hunk 1]
```diff
@@ -80,6 +91,11 @@ roleRef:
   apiGroup: rbac.authorization.k8s.io
   kind: Role
   name: kubevirt-dpdk-checker
+---
+apiVersion: v1
+kind: ServiceAccount
+metadata:
+  name: dpdk-checkup-traffic-gen-sa
```
Comment (orelmisan) [inline modules/virt-checking-cluster-dpdk-readiness.adoc:98]: This part is the main fix for BZ 2187603.

File: modules/virt-checking-cluster-dpdk-readiness.adoc [hunk 2]
```diff
@@ -90,7 +106,56 @@ roleRef:
...
+- system:serviceaccount:dpdk-checkup-ns:dpdk-checkup-traffic-gen-sa
+volumes:
+- hostPath
+----
+
+. Apply the `SecurityContextConstraints` manifest:
++
+[source,terminal]
+----
+$ oc apply -n <target_namespace> -f <dpdk_scc>.yaml
```
Comment (orelmisan) [inline modules/virt-checking-cluster-dpdk-readiness.adoc:155]: SCC is not a namespaced object, so the namespace specification could be dropped:
```suggestion
$ oc apply -f <dpdk_scc>.yaml
```

Comment (orelmisan) [inline modules/virt-checking-cluster-dpdk-readiness.adoc:155]: This part is the main fix for BZ 2187603.

**Review by orelmisan — COMMENTED** [review]
Thank you for the PR @apinnick.
I've marked the parts that IMO are important to fix [BZ 2187603](https://bugzilla.redhat.com/show_bug.cgi?id=2187603).
I suggest to introduce the rest of the changes in a follow-up PR.

**Comment by apinnick** [comment]
@yossisegev Ready for QE review.

File: modules/virt-measuring-latency-vm-secondary-network.adoc [hunk 3]
```diff
@@ -4,13 +4,18 @@
...
 
-You use a predefined checkup to verify network connectivity and measure latency between two virtual machines (VMs) that are attached to a secondary network interface.
+You use a predefined checkup to verify network connectivity and measure latency between two virtual machines (VMs) that are attached to a secondary network interface. The checkup uses the ping utility.
 
-To run a checkup for the first time, follow the steps in the procedure.
+You run a latency checkup by performing the following steps:
 
-If you have previously run a checkup, skip to step 5 of the procedure because the steps to install the framework and enable permissions for the checkup are not required.
+. Create a service account, role, and role bindings to provide cluster access permissions to the latency checkup.
+. Create a config map. The config map provides the input for the framework to run the checkup and stores the results.
```
Comment (orelmisan) [inline modules/virt-measuring-latency-vm-secondary-network.adoc:14]:
```suggestion
. Create a config map. The config map provides the input for the checkup and also stores the results.
```

File: modules/virt-measuring-latency-vm-secondary-network.adoc [hunk 4]
```diff
@@ -4,13 +4,18 @@
...
+. Create a job to run the checkup.
+. Review the results in the config map.
+. Optional: To rerun the checkup, delete the existing job and config map and then create a new job and config map.
```
Comment (orelmisan) [inline modules/virt-measuring-latency-vm-secondary-network.adoc:17]:
```suggestion
. Optional: To rerun the checkup, delete the existing job and config map and then create a new config map and job.
```

**Review by orelmisan — COMMENTED** [review]
Thank you for the changes @apinnick.

**Review by orelmisan — APPROVED** [review]
Thank you for the changes @apinnick

File: modules/virt-measuring-latency-vm-secondary-network.adoc
Comment (yossisegev) [inline modules/virt-measuring-latency-vm-secondary-network.adoc:1]: Why is this file part of this PR?
The PR is supposed to address DPDK checkup documentation, while this file is about the latency checkup.

File: modules/virt-checking-cluster-dpdk-readiness.adoc [hunk 5]
```diff
@@ -80,6 +91,11 @@ roleRef:
   apiGroup: rbac.authorization.k8s.io
   kind: Role
   name: kubevirt-dpdk-checker
+---
+apiVersion: v1
+kind: ServiceAccount
+metadata:
+  name: dpdk-checkup-traffic-gen-sa
 ----
 ====
```
Comment (yossisegev) [inline modules/virt-checking-cluster-dpdk-readiness.adoc:101]: The following line is confusing - the manifest above contains Role, RoleBinding and ServiceAccount, but the line
` Apply the checkup roles manifest:`
implies that only roles should be applied, which gives the impression that the manifest above is not relevant.

File: modules/virt-checking-cluster-dpdk-readiness.adoc [hunk 6]
```diff
@@ -90,7 +106,56 @@ roleRef:
 $ oc apply -n <target_namespace> -f <dpdk_roles>.yaml
```
Comment (yossisegev) [inline modules/virt-checking-cluster-dpdk-readiness.adoc:106]: On the same note - `dpdk_roles` should be renamed to something that reflects all the resources it covers, not only the `Roles`.

**Review by yossisegev — CHANGES_REQUESTED** [review]

File: modules/virt-measuring-latency-vm-secondary-network.adoc
Comment (apinnick) [reply modules/virt-measuring-latency-vm-secondary-network.adoc:1]: This PR includes some general doc housekeeping.

Comment (yossisegev) [reply modules/virt-measuring-latency-vm-secondary-network.adoc:1]: Ack

**Comment by yossisegev** [comment]
I don't see a `MachineConfigPool` definition in this doc.
@orelmisan Please correct me if I'm wrong - we should have a `MachineConfigPool` here, don't we?

**Comment by orelmisan** [comment]
> I don't see a `MachineConfigPool` definition in this doc. @orelmisan Please correct me if I'm wrong - we should have a `MachineConfigPool` here, shouldn't we?
> 
> And I also don't see the SR-IOV resources defined in this doc. Should this doc also cover these, or is this doc only supposed to cover the DPDK checkup job (and its mandatory related resources, e.g. ServioceAccount, Role, etc.), leaving it to the user to determine what would be the underlying infrastructure on the nodes, that would serve the DPDK job and VM (which can be SR-IOV, or other network methodology)?

@yossisegev, there is another PR that describes how to configure the cluster (https://github.com/openshift/openshift-docs/pull/58584).
This document assumes that the cluster is already configured.

File: modules/virt-checking-cluster-dpdk-readiness.adoc [hunk 7]
```diff
@@ -4,19 +4,30 @@
...
+= DPDK checkup
 
-Use a predefined checkup to verify if your {product-title} cluster node can run a virtual machine (VM) with a Data Plane Development Kit (DPDK) workload. The checkup runs traffic between a traffic generator pod and a VM running a test DPDK application, and checks for packet loss.
+Use a predefined checkup to verify that your {product-title} cluster node can run a virtual machine (VM) with a Data Plane Development Kit (DPDK) workload with zero packet loss. The DPDK checkup runs traffic between a traffic generator pod and a VM running a test DPDK application.
+
+You run a DPDK checkup by performing the following steps:
+
+. Create a service account, role, and role bindings for the DPDK checkup and a service account for the traffic generator pod.
+. Create a security context constraints resource for the traffic generator pod.
+. Create a config map. The config map provides the input to run the checkup and stores the results.
```
Comment (mburke5678) [inline modules/virt-checking-cluster-dpdk-readiness.adoc:15]: s/Create a config map. The config map provides/Create a config map to provide (minimalism)

File: modules/virt-measuring-latency-vm-secondary-network.adoc [hunk 8]
```diff
@@ -4,13 +4,18 @@
...
 
-You use a predefined checkup to verify network connectivity and measure latency between two virtual machines (VMs) that are attached to a secondary network interface.
+You use a predefined checkup to verify network connectivity and measure latency between two virtual machines (VMs) that are attached to a secondary network interface. The latency checkup uses the ping utility.
 
-To run a checkup for the first time, follow the steps in the procedure.
+You run a latency checkup by performing the following steps:
 
-If you have previously run a checkup, skip to step 5 of the procedure because the steps to install the framework and enable permissions for the checkup are not required.
+. Create a service account, role, and role bindings to provide cluster access permissions to the latency checkup.
+. Create a config map. The config map provides the input to run the checkup and stores the results.
```
Comment (mburke5678) [inline modules/virt-measuring-latency-vm-secondary-network.adoc:14]: s/Create a config map. The config map provides/Create a config map to provide (minimalism)

File: virt/support/monitoring/virt-monitoring-overview.adoc [hunk 9]
```diff
@@ -9,7 +9,9 @@ toc::[]
 You can monitor the health of your cluster and virtual machines (VMs) with the following tools:
 
 xref:../../../virt/support/monitoring/virt-running-cluster-checkups.adoc#virt-running-cluster-checkups[{product-title} cluster checkup framework]::
-Check network connectivity and latency by using predefined, automated tests.
+Run automated tests on your cluster with the {product-title} cluster checkup framework to check the following:
```
Comment (mburke5678) [inline virt/support/monitoring/virt-monitoring-overview.adoc:12]: Per ISG, you need a descriptive word for the items in the list after _following_.
For ex:

> check the following conditions:

**Comment by yossisegev** [comment]
LGTM (CNV Network QE)

File: virt/support/monitoring/virt-monitoring-overview.adoc (hunk 9 above)
Comment (apinnick) [reply virt/support/monitoring/virt-monitoring-overview.adoc:12]: Thank-you! I knew that I needed a description but I could not think of the right word. :-)

**Pull request merged**